test/controllers/businessactivities/WhatYouNeedControllerSpec.scala

```diff
@@ -85,6 +103,18 @@ class WhatYouNeeControllerSpec extends AmlsSpec with MockitoSugar with ScalaFutu
           doc.getElementById("ba-whatyouneed-button").attr("href") mustBe routes.InvolvedInOtherController.get().url
         }
       }
+      "Redirect to registration progress page" when {
```

Comment:
As stated in the controller. Do we want to do this? Is this error recoverable? If not we likely want to raise an exception such that we can investigate. If we encounter a non recoverable error such as failing to locate business matching in the cache then sending the user back to the progress page might be confusing? The user will keep trying to get into the section and could get stuck in a loop without an exception being thrown therefore giving us nothing to investigate. 

If we need this data to be able to proceed then should we raise an error if this data can't be located? 

The alternative is to load a default if the data can't be retrieved.

Reply:
Done